File: Euclid/Messages/Incoming/WHISPER.cs
using Euclid.Game;
using Euclid.Network.Streams;
using Euclid.Util.Extensions;
using System.Collections.Generic;
using System.Linq;

namespace Euclid.Messages.Incoming
{
    class WHISPER : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            if (!player.Authenticated)
            {
                return;
            }

            if (player.RoomEntity.Room == null)
            {
                return;
            }

            string name = request.GetArgument(0, " ");
            string message = request.Content.Substring(name.Length + 1).FilterInput();

            Player target = name.Length > 0 ? player.RoomEntity.Room.EntityManager.GetEntities<Player>().FirstOrDefault(x => x.EntityData.Name == name) : player;

            if (target == null)
            {
                return;
            }

            player.RoomEntity.Talk(ChatMessageType.WHISPER, message, List.Create(target, player));
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/WHISPER.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.entity.Humanoid;
import euclid.game.util.ChatMessageType;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;
import euclid.util.extensions.StringExtensions;

import java.util.List;

public class WHISPER implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (!player.isAuthenticated()) {
            return;
        }

        if (player.getRoomEntity().getRoom() == null) {
            return;
        }

        String name = request.getArgument(0, " ");
        String message = StringExtensions.filterInput(request.getContent().substring(name.length() + 1));

        List<Humanoid> entities = player.getRoomEntity().getRoom().getEntityManager().getEntities(Humanoid.class);
        Humanoid target = name.length() > 0 ? entities.stream()
                .filter(x -> x.getEntityData().getName().equals(name))
                .findFirst()
                .orElse(null) : null;

        if (target == null) {
            return;
        }

        if (target instanceof Player targetPlayer) {
            player.getRoomEntity().talk(ChatMessageType.WHISPER, message, List.of(targetPlayer));
        }
    }
}
